### PR TITLE
[codegen/python] Fix pyname bug

### PR DESCRIPTION
**Pull request by komalali**
Fixes: https://github.com/pulumi/pulumi-kubernetes/issues/1179

### CODE REVIEW

File: pkg/codegen/python/python.go [hunk 1]
```diff
@@ -119,8 +119,159 @@ func PyName(name string) string {
 				continue
 			}
 
-			// We want to fold digits immediately following an acronym into the same
-			// component as the acronym.
+			// We want to fold digits or the lowercase letter 's' immediately following
+			// an acronym into the same component as the acronym.
+			if unicode.IsDigit(char) || char == 's' {
```
Comment (justinvp) [inline pkg/codegen/python/python.go:124]: Is it possible to share more of the code? Maybe something like:

```go
func PyName(name string) string {
	return pyNameImpl(name, false /*legacy*/)
}

func PyNameLegacy(name string) string {
	return pyNameImpl(name, true /*legacy*/)
}

// Better name ideas?
func pyNameImpl(name string, legacy bool) string {
	...
}
```

File: pkg/codegen/python/python_test.go [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+	{"kubeletConfigKey", "kubelet_config_key"},
+	{"podCIDR", "pod_cidr"},
+	{"podCidr", "pod_cidr"},
+	{"podCIDRs", "pod_cidrs"},
+	{"podCIDRSet", "pod_cidr_set"},
+	{"Sha256Hash", "sha256_hash"},
+	{"SHA256Hash", "sha256_hash"},
+}
+
+func TestPyName(t *testing.T) {
```
Comment (justinvp) [inline pkg/codegen/python/python_test.go:18]: Also add some tests for the legacy behavior to ensure it doesn't change?

File: pkg/codegen/python/python.go [hunk 3]
```diff
@@ -119,8 +119,159 @@ func PyName(name string) string {
...
+			if unicode.IsUpper(char) {
+				// stateLowerOrNumber -> stateUpper
+				if result.Len() != 0 {
+					result.WriteRune('_')
+				}
+				result.WriteString(currentComponent.String())
+
+				currentComponent.Reset()
+				currentComponent.WriteRune(unicode.ToLower(char))
+				state = stateUpper
```
Comment (lblackstone) [inline pkg/codegen/python/python.go:155]: Should this be `unicode.ToUpper(char))`?

File: pkg/codegen/python/python.go (hunk 1 above)
Comment (komalali) [reply pkg/codegen/python/python.go]: Hah yeah that's a much better idea!

File: pkg/codegen/python/python_test.go [hunk 4]
```diff
@@ -0,0 +1,40 @@
+package python
+
+import "testing"
+
+var pyNameTests = []struct {
+	input    string
+	expected string
+	legacy   string
+}{
+	{"kubeletConfigKey", "kubelet_config_key", "kubelet_config_key"},
```
Comment (justinvp) [inline pkg/codegen/python/python_test.go:10]: Mind adding some more cases mentioned in the original bug?

"non_resource_ur_ls": "nonResourceURLs"
"pod_i_ps": "podIPs"
"pod_cid_rs": "podCIDRs"

File: pkg/codegen/python/python_test.go [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+	{"SHA256Hash", "sha256_hash", "sha256_hash"},
+}
+
+func TestPyName(t *testing.T) {
+	for _, tt := range pyNameTests {
+		t.Run(tt.input, func(t *testing.T) {
+			result := PyName(tt.input)
+			if result != tt.expected {
+				t.Errorf("expected \"%s\"; got \"%s\"", tt.expected, result)
+			}
```
Comment (justinvp) [inline pkg/codegen/python/python_test.go:26]: Nit: You could import `"github.com/stretchr/testify/assert"` and do:

```suggestion
			assert.Equal(t, tt.expected, result)
```

File: pkg/codegen/python/python.go (hunk 3 above)
Comment (komalali) [reply pkg/codegen/python/python.go]: This was from the original PyName algo and wasn't causing any issues. I cleaned it up and added some tests which all look good so I believe this is working as expected.

**Pull request merged**